GRDB/Core/Configuration.swift

```diff
@@ -81,6 +81,27 @@ public struct Configuration {
     ///
     /// Default: nil
     public var passphrase: String?
+    
+    // Valid options for the cipher_page_size setting
+    public enum CipherPageSize: Int {
+        case pageSize1K = 1024
+        case pageSize2K = 2048
+        case pageSize4K = 4096
+        case pageSize8K = 8192
+        case pageSize16K = 16384
+        case pageSize32K = 32768
+        case pageSize64K = 65536
+    }
+    
+    /// The cipher_page_size for encrypted databases
+    ///
+    /// Default: .pageSize1K - this corresponds to the default used until now in SQLCipher/GRDBCipher 3
+    public var cipherPageSize: CipherPageSize = .pageSize1K
```

Comment:
[The doc](https://www.zetetic.net/sqlcipher/sqlcipher-api/#cipher_page_size) (for SQLCipher 4) says: "The value for cipher_page_size must be a power of two between 512 and 65536 inclusive." Shouldn't we expose 512, too? What will we do when 128K becomes available? What are the bad consequences of exposing a regular `Int`?

Reply:
The biggest advantage I see to the enum is that we can guarantee that only valid values are used compile time - mostly just less possibilities for potential mistakes. No real reason 512 couldn't be exposed as well (although there is very little reason to use 512). It does mean a slight amount of potential maintenance if they add larger sizes (although I don't think that is likely - the default even with 4.0 is only 4K) - but I think that is fairly minor.

Reply:
All right, then! Don't change anything 👍